Mathlib/Topology/Constructions.lean

```diff
@@ -1053,6 +1054,117 @@ lemma IsClosedEmbedding.sumElim {f : X → Z} {g : Y → Z}
   rw [IsClosedEmbedding.isClosedEmbedding_iff_continuous_injective_isClosedMap] at hf hg ⊢
   exact ⟨hf.1.sumElim hg.1, h, hf.2.2.sumElim hg.2.2⟩
 
+section IsInducing
+
+variable {f : X → Z} {g : Y → Z}
+
+lemma Topology.IsInducing.sumElim_left (h : IsInducing (Sum.elim f g)) : IsInducing f := by
+  rw [← elim_comp_inl f g]
+  exact h.comp IsEmbedding.inl.isInducing
+
+lemma Topology.IsInducing.sumElim_right (h : IsInducing (Sum.elim f g)) : IsInducing g := by
+  rw [← elim_comp_inr f g]
+  exact h.comp IsEmbedding.inr.isInducing
+
+theorem Topology.IsInducing.sumElim (hf : IsInducing f) (hg : IsInducing g)
+    (hFg : Disjoint (closure (range f)) (range g)) (hfG : Disjoint (range f) (closure (range g))) :
+    IsInducing (Sum.elim f g) := by
+  apply isInducing_iff_nhds.mpr
+  intro x
+  apply le_antisymm ((hf.continuous.sumElim hg.continuous).tendsto x).le_comap
+  intro s hs
+  rw [mem_comap_iff_compl, ← image_preimage_inl_union_image_preimage_inr sᶜ,
+    image_union, ← image_comp, ← image_comp, elim_comp_inl, elim_comp_inr,
+    preimage_compl, preimage_compl, compl_union, inter_mem_iff,
+    ← mem_comap_iff_compl, ← mem_comap_iff_compl]
+  rw [← disjoint_principal_nhdsSet] at hFg
+  rw [← disjoint_nhdsSet_principal] at hfG
+  obtain x | x := x <;>
+  simp only [nhds_inl, nhds_inr, mem_map] at hs <;>
+  simp only [elim_inl, elim_inr, ← hf.nhds_eq_comap, ← hg.nhds_eq_comap, hs, true_and, and_true] <;>
+  convert mem_bot <;>
+  rw [comap_eq_bot_iff_compl_range] <;>
+  [(rw [← disjoint_principal_right]
+    apply hfG.mono_left);
+   (rw [← disjoint_principal_left]
+    apply hFg.mono_right)] <;>
+  exact nhds_le_nhdsSet (mem_range_self x)
+
+theorem isInducing_sumElim :
+    IsInducing (Sum.elim f g) ↔ IsInducing f ∧ IsInducing g ∧
+    Disjoint (closure (range f)) (range g) ∧
+    Disjoint (range f) (closure (range g)) := by
+  constructor
```

Comment:
You could combine this with the second case, to something like:
`refine ⟨fun h => ?_, fun ⟨hf, hg, hFg, hfG⟩ => hf.sumElim hg hFg hfG⟩`
and then delete the `intro h` as well as the second case, and un-indent everything.